lib/plug/static.ex

```diff
@@ -179,10 +184,11 @@ defmodule Plug.Static do
 
   defp serve_static({:ok, conn, file_info, path}, segments, options) do
     %{qs_cache: qs_cache, et_cache: et_cache, et_generation: et_generation,
-      gzip?: gzip?, brotli?: brotli?, headers: headers} = options
+      gzip?: gzip?, brotli?: brotli?, headers: headers, mime_types: types} = options
     case put_cache_header(conn, qs_cache, et_cache, et_generation, file_info, path) do
       {:stale, conn} ->
-        content_type = segments |> List.last |> MIME.from_path
+        filename = List.last(segments)
+        content_type = types[filename] || MIME.from_path(filename)
```

Comment:
Small nitpick, but since we know that `mime_types` is a map, I think it would be beneficial to use `Map.get/2` here so that it's more declarative and we'd catch errors sooner. We could even go as far as using `Map.get_lazy(types, filename, fn -> MIME.from_path(filename) end)`, but that's up to the others as well :)